### DISCUSSION TITLE
Serializer validation ignore "required=False" in nested serializer on 3.14 -> 3.15.1 update

### DISCUSSION THREAD

**Discussion by Jeanbouvatt** [encode/django-rest-framework#9420] [Potential Issue] [answered]
I have a pretty huge serializer, which, upon creation, relies on a nested serializer.
My code looks like this:
```
class MainSerializer(ModelSerializer, NestedCreateMixin, NestedUpdateMixin):
    # ...
    my_nested_serializer = SubSerializer(many=True, required=True, read_only=False)

class SubSerializer(ModelSerializer):
    my_field = serializers.PrimaryKeyRelatedField(queryset=MyModel.objects.all(), required=False, read_only=False)
```
which seems good when inspecting the serializer:
```
>>> serializer = self.get_serializer(data=request.data)
>>> serializer
MainSerializer(...):
    my_nested_serializer = SubSerializer(many=True, read_only=False, required=True):
           my_field = PrimaryKeyRelatedField(queryset=<...>, read_only=False, required=False)
````
but...
```
>>> serializer.is_valid()
False
>>> serializer.errors
{'my_nested_serializer': [{'my_field': [ErrorDetail(string='Ce champ est obligatoire.', code='required')]}, {'my_field': [ErrorDetail(string='Ce champ est obligatoire.', code='required')]}]}
```
"Ce champ est obligatoire.'" being the french default error message for a required field not provided.

My code was working as intended, but when updating from 3.14 to 3.15.1 (doing so to use python3.12), it began to fail.

Any Idea? It very much looks like a regression.

**Comment by Jeanbouvatt** [comment, marked as answer]
update: 
it seems that I now have a UniqueTogetherValidator that discard my "required=False".

This UniqueTogetherValidator exists because of my model definition:

```
class Meta:
    constraints = [models.UniqueConstraint(fields=["my_field", "some_other_field"],name="name")]
 ```

If I remember correctly, in 3.14 UniqueConstraint did not create UniqueTogetherValidator automatically, now It does and thus it fails.

**Reply by Jeanbouvatt** [reply]
https://www.django-rest-framework.org/community/3.15-announcement/#primary-support-of-uniqueconstraint

**Reply by Jeanbouvatt** [reply]
TL;DR:  DRF enforce a UniqueTogetherValidator that makes validation fails. I will use https://www.django-rest-framework.org/api-guide/validators/#optional-fields
to make my code work
